addon/services/dwelling.js

```diff
@@ -0,0 +1,118 @@
+import { guidFor } from "@ember/object/internals";
+import { inject as service } from "@ember/service";
+import Dwelling, { DwellingComplete } from "ember-ebau-gwr/models/dwelling";
+
+import GwrService from "./gwr";
+
+export default class DwellingService extends GwrService {
+  @service building;
+
+  cacheKey(dwelling) {
+    return `${guidFor(dwelling)}-${dwelling.EWID}`;
+  }
+  cacheClass = Dwelling;
+
+  async get(EWID, EGID) {
+    if (!EWID) {
```

Comment:
Nitpick: If we check the required `EWID` we could also check the `EGID` in the same go since this is also required.

Reply:
Done